### PR TITLE
Added LineEqationTests.cpp

### PR DESCRIPTION
**Pull request by paarongiroux**
Added tests for LineEquation.cpp
## Motivation and Context

part of transitioning tests to gtest
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/LineEquation/LineEquation.cpp [hunk 1]
```diff
@@ -90,6 +90,7 @@ namespace Isis {
     }
     else if(!p_slopeDefined) {
       p_slope = (p_y[0] - p_y[1]) / (p_x[0] - p_x[1]);
+      p_slopeDefined = true;
```
Comment (jessemapel) [inline isis/src/base/objs/LineEquation/LineEquation.cpp:93]: Why is this change here? Is this related to testing?

Comment (paarongiroux) [reply isis/src/base/objs/LineEquation/LineEquation.cpp:93]: Sorry, forgot about that change.
When writing the tests, I noticed that the flags p_slopeDefined and p_interceptDefined were always false, meaning that the slope and intercept had to be recalculated every time their respective methods were called. This also resulted the methods haveSlope() and haveIntercept() always returning false whether or not the slope and intercept had been defined. I talked with Stuart about it and he said I should just go ahead and change it.

**Comment by jlaura** [comment]
@Kelvinrr I tagged you as a reviewer on this. Sometime next week, can you give this a review? Thanks!

**Comment by Kelvinrr** [comment]
I already looked it over before he PRed it in. I was letting it hang so other eyes other than mine and his can look it over before it's merged.

**Review by Kelvinrr — APPROVED** [review]
